src/cplus_plugin/gui/component_item_model.py

```diff
@@ -186,7 +194,9 @@ def set_layer(self, layer: QgsMapLayer) -> bool:
 
     @staticmethod
     @abstractmethod
-    def create(model_component: BaseModelComponent) -> "ModelComponentItem":
+    def create(
+        model_component: BaseModelComponent, checkable: bool = False
```

Comment:
@osundwajeff kindly see earlier comment on delegating the setting of check status to the item model.